examples/microwave-oven-app/microwave-oven-common/src/microwave-oven-device.cpp

```diff
@@ -0,0 +1,112 @@
+/*
+ *
+ *    Copyright (c) 2023 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+#include "microwave-oven-device.h"
+
+using namespace chip;
+using namespace chip::app;
+using namespace chip::app::Clusters;
+using namespace chip::app::Clusters::OperationalState;
+using namespace chip::app::Clusters::MicrowaveOvenControl;
+using OperationalStateEnum = chip::app::Clusters::OperationalState::OperationalStateEnum;
+using Status               = Protocols::InteractionModel::Status;
+
+Protocols::InteractionModel::Status
+MicrowaveOvenDevice::HandleMicrowaveOvenSetCookingParametersCallback(uint8_t cookMode, uint32_t cookTime, uint8_t powerSetting)
+{
+    // placeholder implementation
+    // TODO: set Microwave Oven cooking mode by cookMode.Value().
```

Comment:
Please create an issue to add in the TODO (e.g. `TODO(#6789012): set Microwave Oven cooking mode by cookMode.Value(`)

Reply:
issue has been created - #30609